cpp/src/groupby/groupby.cu

```diff
@@ -183,6 +184,22 @@ groupby::groups groupby::get_groups(table_view values, rmm::mr::device_memory_re
   }
 }
 
+std::unique_ptr<column> groupby::replace_nulls(column_view const& value,
+                                               cudf::replace_policy const& replace_policy,
+                                               rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  if (value.is_empty()) { return cudf::empty_like(value); }
+  if (not value.has_nulls()) { return std::make_unique<cudf::column>(value); }
+
+  auto group_labels = helper().group_labels(rmm::cuda_stream_default);
+  CUDF_EXPECTS(static_cast<cudf::size_type>(group_labels.size()) == value.size(),
+               "Size mismatch between group labels and value.");
```

Comment:
I think this check should be earlier and ensure `value.size()` is equal to the number of keys in the table the `groupby` object was constructed from.